### PR TITLE
Nav-unification: API/WP-Admin behaviour inconsistency 

### PR DESCRIPTION
**Pull request by BogdanUngureanu**
Fixes https://github.com/Automattic/wp-calypso/issues/56340
Alternative for https://github.com/Automattic/jetpack/pull/21881 
#### Changes proposed in this Pull Request:

* The API endpoint used in Calypso did not had the same behaviour as the logic from WP-Admin core menu that computes the menu links, thus breaking the URL submenus in Calypso for plugins that relied on this behaviour.

#### Jetpack product discussion

N/A
#### Testing instructions:

* Get an Atomic site and use Jetpack Beta and switch to this branch.
* Install the WPML plugin: 175618-pb
* Go to Calypso and make sure that the WPML menu (and submenus) are not broken.
* Since this changes affects the behaviour for all submenus, make a quick smoke test to be sure that the existing menus weren't affected.

### CODE REVIEW

File: projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-admin-menu.php [hunk 1]
```diff
@@ -370,8 +370,14 @@ private function prepare_menu_item_url( $url, $parent_slug = '' ) {
...
 		) {
+			$admin_is_parent = false;
+			if ( ! empty( $parent_slug ) ) {
+				$menu_hook       = get_plugin_page_hook( $parent_slug, 'admin.php' );
+				$admin_is_parent = ! empty( $menu_hook ) || ( ( 'index.php' !== $parent_slug ) && file_exists( WP_PLUGIN_DIR . "/$parent_file" ) && ! file_exists( ABSPATH . "/wp-admin/$parent_file" ) );
+			}
+
 			if (
-				( 'admin.php' !== $parent_file && file_exists( WP_PLUGIN_DIR . "/$parent_file" ) && ! is_dir( WP_PLUGIN_DIR . "/$parent_file" ) ) ||
+				( false === $admin_is_parent && 'admin.php' !== $parent_file && file_exists( WP_PLUGIN_DIR . "/$parent_file" ) && ! is_dir( WP_PLUGIN_DIR . "/$parent_file" ) ) ||
```
Comment (obenland) [inline projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-admin-menu.php:380]: Miguel [had a suggestion](https://github.com/Automattic/jetpack/pull/21881/files#r763178687) to remove the `'admin.php' !== $parent_file`. Should we adopt that here?

Comment (BogdanUngureanu) [reply projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-admin-menu.php]: yes, forgot about that change, thanks!

**Review by obenland — APPROVED** [review]
Tests are green!

**Review by mmtr — APPROVED** [review]

**Comment by BogdanUngureanu** [comment]
Deployed in WPCOM with r236388-wpcom

**Pull request merged**